source/common/http/conn_manager_impl.cc

```diff
@@ -770,6 +786,50 @@ void ConnectionManagerImpl::ActiveStream::refreshCachedRoute() {
   cached_route_.value(std::move(route));
 }
 
+void ConnectionManagerImpl::ActiveStream::encode100ContinueHeaders(
+    ActiveStreamEncoderFilter* filter, HeaderMap& headers) {
+  // Make sure commonContinue continues encode100ContinueHeaders.
+  has_continue_headers_ = true;
+
+  // Similar to the block in encodeHeaders, run encode100ContinueHeaders on each
+  // filter. This is simpler than that case because 100 continue implies no
+  // end-stream, and because there are normal headers coming there's no need for
+  // complex continuation logic.
+  std::list<ActiveStreamEncoderFilterPtr>::iterator entry = commonEncodePrefix(filter, false);
+  for (; entry != encoder_filters_.end(); entry++) {
+    ASSERT(!(state_.filter_call_state_ & FilterCallState::Encode100ContinueHeaders));
+    state_.filter_call_state_ |= FilterCallState::Encode100ContinueHeaders;
+    FilterHeadersStatus status = (*entry)->handle_->encode100ContinueHeaders(headers);
+    state_.filter_call_state_ &= ~FilterCallState::Encode100ContinueHeaders;
+    ENVOY_STREAM_LOG(trace, "encode 100 continue headers called: filter={} status={}", *this,
+                     static_cast<const void*>((*entry).get()), static_cast<uint64_t>(status));
+    if (!(*entry)->commonHandleAfter100ContinueHeadersCallback(status)) {
+      return;
+    }
+  }
+
+  // Stip the T-E headers etc. Defer other header additions as well as drain-close logic to the
+  // continuation headers.
```

Comment:
Do we care that this function will get called twice in the proxy 100 continue case? Now it doesn't matter but might it matter in the future? Might be worth putting some comments in the code to watch out for this.